contracts/root/predicates/ERC20Predicate.sol

```diff
@@ -0,0 +1,299 @@
+      require(
+        participant == address(inputItems[2].toUint()), // from
+        "Withdrawer and referenced tx do not match"
+      );
+      closingBalance = BytesLib.toUint(logData, 64); // output1
+    } else if (eventSignature == LOG_TRANSFER_EVENT_SIG) {
+      // event LogTransfer(
+      //   address indexed token, address indexed from, address indexed to,
+      //   uint256 amountOrTokenId, uint256 input1, uint256 input2, uint256 output1, uint256 output2)
+      if (participant == address(inputItems[2].toUint())) { // A. Participant transferred tokens
+        closingBalance = BytesLib.toUint(logData, 96); // output1
+      } else if (participant == address(inputItems[3].toUint())) { // B. Participant received tokens
+        closingBalance = BytesLib.toUint(logData, 128); // output2
+        oIndex = 1;
+      } else {
+        revert("tx / log doesnt concern the participant");
+      }
+    } else {
+      revert("Exit type not supported");
+    }
+  }
+
+  /**
+   * @notice Process the transaction to start a MoreVP style exit from
+   * @param exitTx Signed exit transaction
+   */
+  function processExitTx(bytes memory exitTx)
+    internal
+    view
+    returns(ExitTxData memory txData)
+  {
+    RLPReader.RLPItem[] memory txList = exitTx.toRlpItem().toList();
+    require(txList.length == 9, "MALFORMED_WITHDRAW_TX");
+    txData.childToken = RLPReader.toAddress(txList[3]); // corresponds to "to" field in tx
+    (txData.signer, txData.txHash) = getAddressFromTx(txList, withdrawManager.networkId());
+    if (txData.signer == msg.sender) { // exit tx is signed by exitor himself
+      (txData.exitAmount, txData.burnt) = processExitTxSender(RLPReader.toBytes(txList[5]));
+    } else {
+      txData.exitAmount = processExitTxCounterparty(RLPReader.toBytes(txList[5]));
+    }
+  }
+
+  function processExitTxSender(bytes memory txData)
+    internal
+    pure
+    returns (uint256 exitAmount, bool burnt)
+  {
+    bytes4 funcSig = BytesLib.toBytes4(BytesLib.slice(txData, 0, 4));
+    if (funcSig == WITHDRAW_FUNC_SIG) {
+      require(txData.length == 36, "Invalid tx"); // 4 bytes for funcSig and a single bytes32 parameter
+      exitAmount = BytesLib.toUint(txData, 4);
+      burnt = true;
+    } else if (funcSig == TRANSFER_FUNC_SIG) {
+      require(txData.length == 68, "Invalid tx"); // 4 bytes for funcSig and a 2 bytes32 parameters (to, value)
+      exitAmount = BytesLib.toUint(txData, 36);
+    } else {
+      revert("Exit tx type not supported");
+    }
+  }
+
+  function processExitTxCounterparty(bytes memory txData)
+    internal
+    view
+    returns (uint256 exitAmount)
+  {
+    require(txData.length == 68, "Invalid tx"); // 4 bytes for funcSig and a 2 bytes32 parameters (to, value)
+    bytes4 funcSig = BytesLib.toBytes4(BytesLib.slice(txData, 0, 4));
+    require(funcSig == TRANSFER_FUNC_SIG, "Only supports exiting from transfer txs");
+    require(
```

Comment:
this info we already have, that's why calling it no ?

Reply:
In the `incomingTransfer` case, this verifies that the counterparty signed a tx to send tokens to the exitor, hence `msg.sender == to` in `transfer(uint256 to, uint256 amount)`